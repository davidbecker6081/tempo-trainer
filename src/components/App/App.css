/* STYLE GUIDE - COLORS ---------------------------------------------- */
/* Main Background - #252525 */
/* Container Background - #303032 */
/* Base Card Background - #252525 */
/* Dark Card Text - #FFFFFF */
/* Title / Highlight - #4EE46F */
/* Box-shadow1 - box-shadow: 0px 0px 2px 2px #4ee46f2b; */
/* Box-shadow2 - box-shadow: 0px 0px 2px 2px #25252563; */
/* Text on Card2 - #181818 */
/* -------------------------------------------------------------------- */

* {
  font-family: 'Oswald', sans-serif;
  font-size: 1em;
}

h1 {
  margin: 0;
  color: #FFFFFF;
  font-size: 2em;
  font-weight: 700;
  text-align: center;
}

h2 {
  margin-top: 0;
  color: #FFFFFF;
  font-size: 2em;
  font-weight: 700;
  letter-spacing: 1.1px;
}

.tempo-trainer {
  color: #4EE46F;
  font-family: 'Racing Sans One', cursive;
  font-size: 5em;
}

.flex-center {
  display: flex;
  align-items: center;
  justify-content: center;
}

.container {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  background-color: #303032;
}

.App {
  display: flex;
}

main {
  flex-direction: column;
  min-height: 100vh;
  padding: 2vh 5vw 2vh 15vw;
  width: 100%;
  background-color: #252525;
  overflow: auto;
}

.effort-average-container {
  justify-content: space-around;
  padding: 2rem;
  margin-bottom: 4rem;
  width: 60vw;
  background-color: #303032;
  border-radius: 5px;
}

@media screen and (max-width: 1150px) {
  * {
    font-size: 14px;
  }

  h1 {
    font-size: 2em;
  }
}

@media screen and (max-width: 770px) {
  * {
    font-size: 12px;
  }

  .App {
    flex-direction: column;
  }

  h1 {
    font-size: 2.5em;
  }

  main {
    padding: 150px 0;
  }

  .effort-average-container {
    flex-direction: column;
  }
}

@media screen and (max-width: 500px) {
  main {
    padding: 215px 0;
  }
}
