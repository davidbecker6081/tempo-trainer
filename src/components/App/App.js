import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import GoogleMap from '../Map/GoogleMap';
import './App.css';

const App = () =>
  (
    <div className="App">
      <header className="App-header">
        <h1 className="App-title">TempoTrainer</h1>
      </header>
      <p className="App-intro">
        To get started, edit <code>src/App.js</code> and save to reload.
      </p>
      <GoogleMap />
    </div>
  );

export default App;
