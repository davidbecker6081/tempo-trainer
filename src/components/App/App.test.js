import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  let mockState;

  beforeEach(() => {
    wrapper = shallow(<App />);
    mockState = {
      range: [],
      channelSet: '',
      originalRange: [],
    };
  });

  it('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });
  describe('range is defined', () => {
    beforeEach(() => { mockState.range = [0, 86]; });

    it('should render correctly', () => {
      wrapper.setState(mockState);
      expect(wrapper.update()).toMatchSnapshot();
    });
  });
  describe('originalRange is defined', () => {
    beforeEach(() => { mockState.originalRange = [0, 86]; });

    it('should render correctly', () => {
      wrapper.setState(mockState);
      expect(wrapper.update()).toMatchSnapshot();
    });
  });
});
