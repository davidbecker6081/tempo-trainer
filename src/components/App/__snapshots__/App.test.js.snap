// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App originalRange is defined should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavigationView />,
        <Route
          render={[Function]}
          to="/"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavigationView />,
          <Route
            render={[Function]}
            to="/"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App range is defined should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavigationView />,
        <Route
          render={[Function]}
          to="/"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavigationView />,
          <Route
            render={[Function]}
            to="/"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
